File: frontend/workiez/src/models/UserDTO.ts
import { District } from "./Enums/DistrictEnum";
import { Gender } from "./Enums/GenderEnum";
import { Province } from "./Enums/ProvinceEnum";
import { Role } from "./Enums/RoleEnum";
import { User } from "./user";

export interface UserDTO {

    userdId:number,
    firstname:string,
    lastname:string,
    email:string,
    username: string,
    phone:string,
    role:Role,
    district:District,
    province:Province
    gender:Gender,
    city:string,
}

File: backend/workiez/src/main/java/workiez/workiez/user/UserDTO.java
package workiez.workiez.user;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long userId;
    private String firstname;
    private String lastname;
    private String username;
    private String email;
    private String phone;
    private Role role;
    private Gender gender;
    private District district;
    private Province province;
    private String city;
}
